cmd/gaia/init/gentx.go

```diff
@@ -86,8 +93,45 @@ following delegation and commission default parameters:
 					return err
 				}
 			}
-			// Run gaiad tx create-validator
+
+			// Set flags for creating gentx
 			prepareFlagsForTxCreateValidator(config, nodeID, ip, genDoc.ChainID, valPubKey)
+
+			// Fetch the amount of coins staked
+			amount := viper.GetString(cli.FlagAmount)
+			coins, err := sdk.ParseCoins(amount)
+			if err != nil {
+				return err
+			}
+
+			inGenesis := false
+			bondDenom := genesisState.StakeData.Params.BondDenom
+
+			// Check if the account is in genesis
+			for _, acc := range genesisState.Accounts {
+				// Ensure that account is in genesis
+				if acc.Address.Equals(key.GetAddress()) {
+
+					// Ensure account contains enough funds of default bond denom
+					if coins.AmountOf(bondDenom).GT(acc.Coins.AmountOf(bondDenom)) {
+						return fmt.Errorf(
+							"Account %s is in genesis, but the only has %s%s available to stake, not %s%s",
+							key.GetAddress(), acc.Coins.AmountOf(bondDenom), bondDenom, coins.AmountOf(bondDenom), bondDenom,
+						)
+					}
+					inGenesis = true
+					break
+				}
+			}
+
+			if !inGenesis {
+				return fmt.Errorf(
```

Comment:
We can eliminate use of `inGenesis` by just moving this error into the above loop (which should also probably get moved to it's own funcion per my previous comment)